core/platform/storage/cloud_storage_emulator_test.py

```diff
@@ -0,0 +1,153 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for cloud_storage_emulator."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import pickle
+
+from core.platform.storage import cloud_storage_emulator
+from core.tests import test_utils
+
+
+class BlobUnitTests(test_utils.TestBase):
+    """Tests for Blob."""
+
+    def test_init_blob_with_str_creates_blob(self):
+        blob = cloud_storage_emulator.Blob('name', 'string', 'png')
+        self.assertEqual(blob.name, 'name')
+        self.assertEqual(blob.download_as_bytes(), b'string')
+        self.assertEqual(blob.content_type, 'png')
+
+    def test_init_blob_with_bytes_creates_blob(self):
+        blob = cloud_storage_emulator.Blob('name', b'string', 'png')
+        self.assertEqual(blob.name, 'name')
+        self.assertEqual(blob.download_as_bytes(), b'string')
+        self.assertEqual(blob.content_type, 'png')
+
+    def test_create_copy(self):
+        orig_blob = cloud_storage_emulator.Blob('name', 'string', 'png')
+        copy_blob = cloud_storage_emulator.Blob.create_copy(orig_blob, 'new')
+        self.assertNotEqual(orig_blob, copy_blob)
+        self.assertNotEqual(orig_blob.name, copy_blob.name)
+        self.assertEqual(
+            orig_blob.download_as_bytes(), copy_blob.download_as_bytes())
+        self.assertEqual(orig_blob.content_type, copy_blob.content_type)
+
+    def test_compare_blob_and_int_is_false(self):
+        orig_blob = cloud_storage_emulator.Blob('name', 'string', 'png')
+        self.assertFalse(orig_blob == 1)
+
+    def test_repr(self):
+        orig_blob = cloud_storage_emulator.Blob('name', 'string', 'png')
+        self.assertEqual(
+            orig_blob.__repr__(), 'Blob(name=name, content_type=png)')
+
+
+class CloudStorageEmulatorUnitTests(test_utils.TestBase):
+    """Tests for CloudStorageEmulator."""
+
+    def setUp(self):
+        super(CloudStorageEmulatorUnitTests, self).setUp()
+        self.emulator = cloud_storage_emulator.CloudStorageEmulator()
+        self.emulator.namespace = 'namespace'
+        self.emulator.reset()
+        self.blob1 = cloud_storage_emulator.Blob(
+            '/file/path.png', b'data', 'png')
+        self.blob2 = cloud_storage_emulator.Blob(
+            '/file/path2.png', b'data2', 'png')
+        self.blob3 = cloud_storage_emulator.Blob(
+            '/different/path.png', b'data2', 'png')
+
+    def tearDown(self):
+        super(CloudStorageEmulatorUnitTests, self).tearDown()
+        self.emulator.reset()
+
+    def test_get_blob(self):
```

Comment:
Please update all these names to be clearer about what each test is testing. Feel free to split each test into multiple parts if needed. (See https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests for documentation on writing test names.)

Reply:
Done.